src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -610,7 +610,10 @@ static async Task OpenCore(
                 (gssEncMode == GssEncryptionMode.Prefer ||
                 // Auth error with/without SSL
                 (e is PostgresException { SqlState: PostgresErrorCodes.InvalidAuthorizationSpecification } &&
-                 (sslMode == SslMode.Prefer && conn.IsSslEncrypted || sslMode == SslMode.Allow && !conn.IsSslEncrypted)))
+                 (sslMode == SslMode.Prefer && conn.IsSslEncrypted || sslMode == SslMode.Allow && !conn.IsSslEncrypted)) ||
+                // Auth error ConnectionReset on Windows
+                (e is NpgsqlException { InnerException: IOException { InnerException: SocketException { SocketErrorCode: SocketError.ConnectionReset } } }
+                 && sslMode == SslMode.Prefer && conn.IsSslEncrypted))
```

Comment:
You can just replace everything here with `sslMode == SslMode.Prefer && conn.IsSslEncrypted || sslMode == SslMode.Allow && !conn.IsSslEncrypted`. Libpq doesn't discriminate for whatever reason the connection failed, so we might as well do the same, as long as we `sslMode` allows that. Might be a good idea to add trace/debug logging, but that's probably something that's better to do in a separate pr.

Reply:
How to test the `SslMode.Allow` part? I could not get it into this state.

Reply:
Again, we do not need either `e is PostgresException { SqlState: PostgresErrorCodes.InvalidAuthorizationSpecification }` or `e is NpgsqlException { InnerException: IOException { InnerException: SocketException { SocketErrorCode: SocketError.ConnectionReset } } })`, there is no need to be specific. If there is any exception, we retry as long as `SslMode` allows that.

```csharp
catch (Exception e) when
	// Any error after trying with GSS or SSL encryption
	(gssEncMode == GssEncryptionMode.Prefer ||
	 sslMode == SslMode.Prefer && conn.IsSslEncrypted ||
	 sslMode == SslMode.Allow && !conn.IsSslEncrypted)
{
```

> How to test the SslMode.Allow part? I could not get it into this state.

Have pg_hba with `hostssl` only.